syft/workers/base.py

```diff
@@ -112,7 +112,14 @@ def __init__(
         self.id = id
         self.is_client_worker = is_client_worker
         self.log_msgs = log_msgs
-        self.verbose = verbose
+        if hasattr(hook, "verbose") and hook.verbose is True:
+            self.verbose = True
+        else:
+            self.verbose = verbose
+
+        if id != "me" and isinstance(hook, sy.TorchHook) and hasattr(hook, "_virtual_workers"):
+            hook._virtual_workers.append(self)
```

Comment:
Wanna make _virtual_workers a `set()` to avoid `if id != "me" ` ?

Reply:
Done!